### PR TITLE
Added timeout retry and handle symbols in record metric

### PR DESCRIPTION
**Pull request by awaage**
Two updates:

- Handles timeouts and will retry
- Handles symbols in the record metric

### CODE REVIEW

File: test/plugin/test_out_graphite.rb [hunk 1]
```diff
@@ -94,6 +94,24 @@ def test_format_metrics
...
+  def test_format_metrics_symbols
+    record = {
+      :'hostname'                  => 'localhost.localdomain',
+      :'dstat.total cpu usage.usr' => '0.0',
+      :'dstat.total cpu usage.sys' => '0.0',
+      :'dstat.total cpu usage.idl' => '100.0',
+      :'dstat.total cpu usage.wai' => '0.0',
+      :'dstat.total cpu usage.hiq' => '0.0',
+      :'dstat.total cpu usage.siq' => '0.0'
+    }
```
Comment (studio3104) [inline test/plugin/test_out_graphite.rb]: Could you follow [this](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#hashes)?

Comment (awaage) [inline test/plugin/test_out_graphite.rb]: I can change it. please be specific.

**Comment by awaage** [comment]
I agree that the input plugin should follow the spec for String metrics.

**Comment by studio3104** [comment]
Then, could you remove implementations to handle symbol problem from this patch?

**Comment by awaage** [comment]
I would suggest keeping the changes. The problem is that there's no official review for input plugins.  Would you rather the plugin fail ? Most people won't have the patience to look into why...

File: test/plugin/test_out_graphite.rb (hunk 1 above)
Comment (studio3104) [inline test/plugin/test_out_graphite.rb]: Oops, please ignore the previous mention, I've overlooked that the keys contain space...

**Comment by studio3104** [comment]
OK, so let's keep the changes.
I'll merge and release if you rebase commits.

**Comment by awaage** [comment]
@studio3104 ping  on this. I rebased commits and there's a new PR

https://github.com/studio3104/fluent-plugin-graphite/pull/7

LMK if any feedback. thanks

**Pull request closed without merging**